code/modules/unit_tests/say.dm

```diff
@@ -21,3 +21,25 @@
 
 	TEST_ASSERT(!expected_mods.len,
 		"Some message mods were expected, but were not returned by get_message_mods: [json_encode(expected_mods)]. Message: [message]")
+
+/// Test to verify COMSIG_MOB_SAY is sent the exact same list as the message args, as they're operated on
+/datum/unit_test/say_signal
+	var/test_message = "Make sure the say signal gets the arglist say is past, no copies!"
+
+/datum/unit_test/say_signal/Run()
+	var/mob/living/dummy = allocate(/mob/living)
+
+	RegisterSignal(dummy, COMSIG_MOB_SAY, .proc/check_say)
+	dummy.say(test_message)
+
+/datum/unit_test/say_signal/proc/check_say(mob/living/source, list/say_args)
+	SIGNAL_HANDLER
+
+	TEST_ASSERT_EQUAL(say_args[SPEECH_MESSAGE], test_message, "Say signal's first arg of the arglist wasn't the expected message.")
+
+	TEST_ASSERT_EQUAL(REF(say_args), source.last_say_args_ref, "Say signal didn't get the argslist of say as a reference. \
```

Comment:
The juggling here shouldn't be necessary. if `REF(say_args) == source.last_say_args_ref`, then the lists would equal each other too.